plugins/scaffolder-backend/src/scaffolder/actions/builtin/publish/github.ts

```diff
@@ -35,13 +31,7 @@ export function createPublishGithubAction(options: {
   config: Config;
 }) {
   const { integrations, config } = options;
-
-  const credentialsProviders = new Map(
-    integrations.github.list().map(integration => {
-      const provider = GithubCredentialsProvider.create(integration.config);
-      return [integration.config.host, provider];
-    }),
-  );
+  const octokitProvider = new OctokitProvider(integrations);
```

Comment:
this makes me wonder if this should have been

```ts
const octokitProvider = OctokitProvider.fromConfig(config);
```

and being passed into the create-function from the outside. Other parts of the code typically don't need to use the integrations, and the fact that the provider is implemented in terms of integrations could be hidden from the caller. Just a thought. This does require us to change the public contract of how actions are created of course ... which is a bigger change. Maybe that's a topic for some other day.

Reply:
I think, that the biggest advantage of creating it in `router.ts` is caching credentials across multiple actions. However, the biggest disadvantage I see is leaking GitHub specific implementation into `router.ts`. I am not convinced it is worth it.

Reply:
Yeah in the current structure of things, I think you are right. In the future we might break out all of the provider specific stuff into separate modules instead of having them in the backend itself, and then it may make more sense. Then, a user would opt into using github in the first place, and actively choosing to add actions for it already in code. But that's for another day